### PR TITLE
Examples gallery

### LINKED ISSUE #69: Run and check examples on CI

### PR DESCRIPTION
**Pull request by ziw-liu**
Render the example scripts in a new 'Examples' section of the documentation site with [sphinx-gallery](https://sphinx-gallery.github.io/stable/index.html).

~Temporarily disabled multi-version build since sphinx-multiversion cannot execute code properly at build time, and we don't have any stable versions yet. One solution when we revisit this after releasing v0.1.0 is to loop over tagged versions in the make file.~
Edit: Build fails when running `make publish` locally on this branch because `examples/README.rst` does not exist in `main` yet.

Fix #69 as the scripts are executed in CI except for the too expensive ones (e.g. writing large files or viewing in napari).

Developers: for guide on writing and building the new examples, see the updated CONTRIBUTING file.

### CODE REVIEW

**Comment by JoOkuma** [comment]
@ziw-liu
I got these warnings when building the documentation

```
generating gallery for auto_examples... [100%] run_single_fov_ome_zarr.py       
computation time summary:
WARNING: The version switcher "file:///home/jordao/Softwares/iohub/docs/build/html//main/_static/switcher.json" file cannot be read due to the following error:
FileNotFoundError(2, 'No such file or directory')
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 12 source files that are out of date
updating environment: [extensions changed ('2')] 20 added, 0 changed, 3 removed
reading sources... [100%] index                                                 
/home/jordao/Softwares/iohub/docs/source/auto_examples/run_coordinate_transform.rst:187: WARNING: Content block expected for the "note" directive; none found.
/home/jordao/Softwares/iohub/docs/source/auto_examples/run_coordinate_transform.rst:188: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/jordao/Softwares/iohub/docs/source/auto_examples/run_large_array_ome_zarr.rst:30: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/jordao/Softwares/iohub/docs/source/auto_examples/run_large_array_ome_zarr.rst:76: WARNING: Explicit markup ends without a blank line; unexpected unindent.
```


Everything else worked great! The examples are nicely displayed.

I noticed that ClearControl's example follows a different format (my bad). Is it worth making it similar to the others?

napari package's examples follows a pattern where they are all executed but without calling the `napari.run` to block the execution.  Is that worth doing for our napari examples rather than not running them?

File: docs/examples/run_coordinate_transform.py [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
+OME-Zarr dataset with two FOV in a single well with different
+coordinate transformations (i.e translation and scaling)
+"""
+
 # %%
-# This script writes two positions using the high content screening (HCS)
-# OME-Zarr dataset with two FOV in a single well with different
-# coordinate transformations (i.e translation and scaling)
+# Setup
+# -----
```
Comment (JoOkuma) [inline docs/examples/run_coordinate_transform.py:12]: Is there any reason for the `Setup/-----` as a header? This is the only example with this

**Comment by ziw-liu** [comment]
> I got these warnings when building the documentation

The version switcher warning is a bug in the theme (cannot load json from local files properly) but works fine when deployed.

I fixed the markup warnings.

File: docs/examples/run_coordinate_transform.py (hunk 1 above)
Comment (ziw-liu) [reply docs/examples/run_coordinate_transform.py:12]: Not really... Will remove for consistency.

**Comment by JoOkuma** [comment]
> Is there any way of testing at least a part of that without taking on napari as a [doc] dependency and dealing with GH Linux runner's Qt issues?

I did some changes to this branch on #154, what do you think?

**Comment by ziw-liu** [comment]
Credit to @AhmetCanSolak for suggesting this point in a [meeting](https://github.com/czbiohub-sf/iohub/wiki/2023-03-06-meeting-notes) (agenda item 4).

**Comment by ziw-liu** [comment]
Test failing because of #160

**Comment by JoOkuma** [comment]
@ziw-liu This bug wasn't created in this PR, but I think you could fix it.

The `make help` command doesn't reflect the current functionalities of the make file.

```
$ make
Please use `make <target>' where <target> is one of
  html       to make standalone HTML files
```
Make file doesn't have `html` command, it only has help, clean, build, and publish.
```
$ make html
make: *** No rule to make target 'html'.  Stop.
```

**Comment by JoOkuma** [comment]
Everything else is great.
I didn't have time to improve the ClearControl example as you suggested. We can open an issue about it if you prefer.

**Comment by ziw-liu** [comment]
> I didn't have time to improve the ClearControl example as you suggested. We can open an issue about it if you prefer.

Done in https://github.com/czbiohub-sf/iohub/pull/153/commits/6b03c5f12bc2f4ae048ad8dd86d54d909af21f40 and https://github.com/czbiohub-sf/iohub/pull/153/commits/8a5d754fa96c070c9be53dd6fc361ab9e7bd5a8e.

**Comment by ziw-liu** [comment]
> Just a sanity check, napari isn't available on the CI, right? So it won't open the viewer and block the execution.

Yes, it will just `pass` for that block.

**Pull request merged**